pkg/proxy/iptables/proxier.go

```diff
@@ -459,17 +459,27 @@ func (proxier *Proxier) syncProxyRules() {
 	glog.V(3).Infof("Syncing iptables rules")
 
 	// Ensure main chains and rules are installed.
-	inputChains := []utiliptables.Chain{utiliptables.ChainOutput, utiliptables.ChainPrerouting}
+	if _, err := proxier.iptables.EnsureChain(utiliptables.TableFilter, iptablesServicesChain); err != nil {
+		glog.Errorf("Failed to ensure that filter chain %s exists: %v", iptablesServicesChain, err)
+		return
+	}
+	if _, err := proxier.iptables.EnsureChain(utiliptables.TableNAT, iptablesServicesChain); err != nil {
+		glog.Errorf("Failed to ensure that nat chain %s exists: %v", iptablesServicesChain, err)
+		return
+	}
 	// Link the services chain.
+	chain := utiliptables.ChainOutput
+	comment := "kubernetes service portals"
+	args := []string{"-m", "comment", "--comment", comment, "-j", string(iptablesServicesChain)}
```

Comment:
also de-dup here.  This comes down to a loop over a table of {table, chain} and all have the same body.